ssm.go

```diff
@@ -78,18 +79,17 @@ func (c DefaultSSMConnector) fetchParametersByNames(client *ssm.SSM, names []str
 	return params, nil
 }
 
-func newSSMClient(retries int) (*ssm.SSM, error) {
-	sess, err := session.NewSessionWithOptions(session.Options{
-		SharedConfigState: session.SharedConfigEnable,
-	})
-	if err != nil {
-		return nil, fmt.Errorf("failed to start session: %w", err)
+func newSSMClient(ctx context.Context, retries int) (*ssm.Client, error) {
+	opts := []func(*config.LoadOptions) error{}
+
+	if 0 < retries {
+		opts = append(opts, config.WithRetryMaxAttempts(retries))
 	}
 
-	// config.MaxRetries is defaults to -1
-	if retries <= 0 {
-		retries = -1
```

Comment:
On v2, default retries is `3`
https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/retries-timeouts/#standard-retryer